mipengine/algorithms/ttest_independent.py

```diff
@@ -0,0 +1,180 @@
+    diff = sum(x1 - x2)
+    diff_sqrd = sum((x1 - x2) ** 2)
+    x1_sqrd_sum = sum(x1**2)
+    x2_sqrd_sum = sum(x2**2)
+
+    sec_transfer_ = {
+        "n_obs_x1": {"data": n_obs_x1, "operation": "sum", "type": "int"},
+        "n_obs_x2": {"data": n_obs_x2, "operation": "sum", "type": "int"},
+        "sum_x1": {"data": x1_sum.item(), "operation": "sum", "type": "float"},
+        "sum_x2": {"data": x2_sum.item(), "operation": "sum", "type": "float"},
+        "diff": {"data": diff.tolist(), "operation": "sum", "type": "float"},
+        "diff_sqrd": {"data": diff_sqrd.tolist(), "operation": "sum", "type": "float"},
+        "x1_sqrd_sum": {
+            "data": x1_sqrd_sum.tolist(),
+            "operation": "sum",
+            "type": "float",
+        },
+        "x2_sqrd_sum": {
+            "data": x2_sqrd_sum.tolist(),
+            "operation": "sum",
+            "type": "float",
+        },
+    }
+
+    return sec_transfer_
+
+
+@udf(
+    sec_local_transfer=secure_transfer(sum_op=True),
+    alpha=literal(),
+    alternative=literal(),
+    return_type=[transfer()],
+)
+def global_independent(sec_local_transfer, alpha, alternative):
+    import scipy.stats as st
+    from scipy.stats import t
+
+    n_obs_x1 = sec_local_transfer["n_obs_x1"]
+    n_obs_x2 = sec_local_transfer["n_obs_x2"]
+    sum_x1 = sec_local_transfer["sum_x1"]
+    sum_x2 = sec_local_transfer["sum_x2"]
+    diff_sum = sec_local_transfer["diff"]
+    diff_sqrd_sum = sec_local_transfer["diff_sqrd"]
+    x1_sqrd_sum = sec_local_transfer["x1_sqrd_sum"]
+    x2_sqrd_sum = sec_local_transfer["x2_sqrd_sum"]
+
+    n_obs = n_obs_x1 + n_obs_x2
+    mean_x1 = sum_x1 / n_obs_x1
+    mean_x2 = sum_x2 / n_obs_x2
+    devel_x1 = x1_sqrd_sum - 2 * mean_x1 * sum_x1 + (mean_x1**2) * n_obs_x1
+    devel_x2 = x2_sqrd_sum - 2 * sum_x2 * mean_x2 + (mean_x2**2) * n_obs_x2
+    sd_x2 = numpy.sqrt(devel_x2 / (n_obs_x2 - 1))
+    sd_x1 = numpy.sqrt(devel_x1 / (n_obs_x1 - 1))
+
+    # standard deviation of the difference between means
+    # sd_x1 = numpy.sqrt((diff_sqrd_sum - (diff_sum**2 / n_obs_x1)) / (n_obs_x1 - 1))
+    # sd_x2 = numpy.sqrt((diff_sqrd_sum - (diff_sum**2 / n_obs_x2)) / (n_obs_x2 - 1))
+
+    # standard error of the difference between means
+    sed_x1 = sd_x1 / numpy.sqrt(n_obs_x1)
+    sed_x2 = sd_x2 / numpy.sqrt(n_obs_x2)
+
+    # standard error on the difference between the samples
+    sed = numpy.sqrt(sed_x1**2.0 + sed_x2**2.0)
+    # t-statistic
+    t_stat = (mean_x1 - mean_x2) / sed
+    df = n_obs_x1 + n_obs_x2 - 2
+
+    # Difference of means
+    diff_mean = mean_x1 - mean_x2
+
+    # Confidence intervals !WARNING: The ci values are not tested. The code should not be modified, unless there is
+    # a test for the new method.
+    if n_obs < 30:
+        ci_lower, ci_upper = t.interval(
+            alpha=1 - alpha, df=df, loc=diff_mean, scale=sed
+        )
+    else:
+        ci_lower, ci_upper = st.norm.interval(alpha=1 - alpha, loc=diff_mean, scale=sed)
```

Comment:
You can safely use the exact form `t.interval` instead of the normal approximation for any `n_obs`. `scipy` handles it internally.